src/app/pages/common/entity/entity-form/components/form-checkbox/form-checkbox.component.html

```diff
@@ -1,8 +1,8 @@
 <div id="{{config.name}}" class="dynamic-field form-checkbox form-element" [formGroup]="group" [ngClass]="fieldShow" [class.has-tooltip]="config.tooltip" *ngIf="!config['isHidden']">
-	<mat-checkbox color="primary" *ngIf="!config.updater" [formControlName]="config.name" [required]="config.required" ix-auto ix-auto-type="checkbox" ix-auto-identifier="{{config.placeholder}}">
+	<mat-checkbox color="primary" *ngIf="!config.updater" [formControlName]="config.name" (click)="(config.readonly && preventClick($event))" [required]="config.required" ix-auto ix-auto-type="checkbox"  ix-auto-identifier="{{config.placeholder}}">
```

Comment:
I'm using the way JavaScript handles logical operators AND and OR to conditionally bind or not bind the `preventClick `method. If config.readonly is true, JavaScript doesn't execute the second part of the && operator since it doesn't need to. Same kind of logic for the optional binding with the OR operator below.

Reply:
And I needed the second part of the logical operator to return true to apply the logic I was going for. That's why it's in a method that always returns true.

Reply:
Makes sense! Thanks for the explanation